### ISSUE TITLE
Implement train.py and configs

### ISSUE THREAD

**Issue by ssenan** [pinellolab/DNA-Diffusion#35]
Create training script and configs using pytorch lightning and hydra

**Comment by ssenan** [comment]
I have created an initial script on my own fork that likely will need some refactoring:
https://github.com/ssenan/DNA-Diffusion/tree/codebase/src
config changes located at: 
https://github.com/ssenan/DNA-Diffusion/tree/codebase/src/configs

After looking at some examples of code using hydra it appears that common practice is to split the config files into several different folders so that they can easily be located an altered for different training runs.  I was wondering if we are interested in following that type of structure?

An example of the config folder styling is located here:
https://github.com/ashleve/lightning-hydra-template/tree/main/configs

Another observation is that parse args for things like seed and learning rate may not be necessary when using hydra as it can easily be altered within the various configs or changing on cli when desired. 

Lastly, I'm unsure of all the logging methods that we aim to use, so my implementation includes a few hypothetical logging methods (These can also be altered and details within the config folders)
